src/serac/physics/thermomechanics_monolithic.hpp

```diff
@@ -0,0 +1,1102 @@
+      tuple<parameter_space...> types{};
+      for_constexpr<sizeof...(parameter_space)>([&](auto i) {
+        parameters_.emplace_back(mesh_, get<i>(types), detail::addPrefix(name_, parameter_names[i]));
+
+        trial_spaces[i + NUM_STATE_VARS] = &(parameters_[i].state->space());
+      });
+    }
+
+    residual_T_ = std::make_unique<
+        ShapeAwareFunctional<shape_trial, scalar_test(scalar_trial, vector_trial, parameter_space...)>>(
+        shape_space, test_space_1, trial_spaces);
+    residual_u_ = std::make_unique<
+        ShapeAwareFunctional<shape_trial, vector_test(scalar_trial, vector_trial, parameter_space...)>>(
+        shape_space, test_space_2, trial_spaces);
+
+    block_thermomech_offsets_.SetSize(NUM_STATE_VARS + 1);
+    block_thermomech_offsets_[0] = 0;
+    block_thermomech_offsets_[1] = temperature_.space().TrueVSize();
+    block_thermomech_offsets_[2] = displacement_.space().TrueVSize();
+    block_thermomech_offsets_.PartialSum();
+
+    block_thermomech_ = std::make_unique<mfem::BlockVector>(block_thermomech_offsets_);
+
+    block_thermomech_->GetBlock(0) = temperature_;
+    block_thermomech_->GetBlock(1) = displacement_;
+
+    block_thermomech_adjoint_ = std::make_unique<mfem::BlockVector>(block_thermomech_offsets_);
+    block_thermomech_adjoint_->GetBlock(0) = temperature_adjoint_;
+    block_thermomech_adjoint_->GetBlock(1) = displacement_adjoint_;
+
+    nonlin_solver_->setOperator(block_residual_with_bcs_);
+
+    block_nonlinear_oper_ = std::make_unique<mfem::BlockOperator>(block_thermomech_offsets_);
+    block_nonlinear_oper_transpose_ = std::make_unique<mfem::BlockOperator>(block_thermomech_offsets_);
+
+    shape_displacement_ = 0.0;
+    initializeThermoMechanicsStates();
+  }
+
+  /// @brief Destroy the ThermoMechanics Functional object
+  virtual ~ThermoMechanicsMonolithic() {}
+
+  /**
+   * @brief Non virtual method to reset temperature and displacement states to zero.  This does not reset design
+   * parameters or shape.
+   *
+   */
+  void initializeThermoMechanicsStates()
+  {
+    temperature_ = 0.0;
+    displacement_ = 0.0;
+
+    temperature_adjoint_ = 0.0;
+    displacement_adjoint_ = 0.0;
+
+    temperature_adjoint_load_ = 0.0;
+    displacement_adjoint_load_ = 0.0;
+  }
+
+  /**
+   * @brief Virtual implementation of required state reset method
+   *
+   */
+  void resetStates(int cycle = 0, double time = 0.0) override
+  {
+    BasePhysics::initializeBasePhysicsStates(cycle, time);
+    initializeThermoMechanicsStates();
+  }
+
+  /**
+   * @brief Set essential temperature boundary conditions (strongly enforced)
+   *
+   * @param[in] temp_bdr The boundary attributes on which to enforce a temperature
+   * @param[in] temp The prescribed boundary temperature function
+   *
+   * @note This should be called prior to completeSetup()
+   */
+  void setTemperatureBCs(const std::set<int>& temp_bdr, std::function<double(const mfem::Vector& x, double t)> temp)
```

Comment:
I think this version should be removed. It's redundant with the following method, right? Or is it needed for some reason?